src/nanovg.c

```diff
@@ -1398,7 +1407,7 @@ static void nvg__flattenPaths(NVGcontext* ctx)
 		}
 
 		// Enforce winding.
-		if (path->count > 2) {
+		if (state->lineStyle == NVG_LINE_SOLID && path->count > 2) {
```

Comment:
For dotted and dashed lines we don't want to reverse the polygon because that will cause the glyphs to jump around as the curve changes shape.